### PR TITLE
Use ghooks for git hooks

### PR DESCRIPTION
**Pull request by ryanlntn**
## Please verify the following:
- [✔] Everything works on iOS/Android
- [✔] `ignite-base` **ava** tests pass
- [✔] `fireDrill.sh` passed

## Describe your PR
Addresses https://github.com/infinitered/ignite/issues/192

Uses the ghooks npm package to implement pre-commit linting. Removes previous git hooks implementation.

@GantMan Please review.

### CODE REVIEW

**Comment by GantMan** [comment]
sorry for the delay, will review soon.

File: ignite-base/package.json [hunk 1]
```diff
@@ -76,5 +77,10 @@
       "XMLHttpRequest",
       "FormData"
     ]
+  },
+  "config": {
+    "ghooks": {
+      "pre-commit": "npm run lint"
```
Comment (GantMan) [inline ignite-base/package.json]: should it lint everything or just the files that are staged?  The previous `lint.sh` would just lint staged files.   Open to thoughts.

**Comment by GantMan** [comment]
2 things @ryanlntn - Firstly, what do you think about adjusting this to lint only staged files of the commit, instead of all files. 

Second, you should update the README that gets generated:
https://github.com/infinitered/ignite/blob/master/ignite-base/README.md.template

**Comment by ryanlntn** [comment]
@GantMan Done and done. 😄

**Comment by GantMan** [comment]
Ryan, I'm trying to test this branch, and I get the following:
image

As you can see, it's not working very well for me.   Can you verify?

**Comment by GantMan** [comment]
@ryanlntn did you verify this fixes the issue?

**Comment by ryanlntn** [comment]
@GantMan Yeah. The only issue you might have is if you don't have snazzy installed globally it will complain when working on the repo. This shouldn't be a problem for endusers.

**Comment by GantMan** [comment]
@ryanlntn - more problems.
It's complaining even though the file is legit.   Lint works fine, but I'm thinking the problem might be that it can't understand that it is ECMA2015
image

My guess is normally it runs in the same folder with a proper `.babelrc` and now it's not happy.   Thoughts?

**Comment by GantMan** [comment]
NOTE: We may have to go down this road: https://github.com/julien-f/babel-standard or something similar.

**Comment by ryanlntn** [comment]
@GantMan Any reason we couldn't duplicate the .babelrc in the project root?

**Comment by GantMan** [comment]
I don't see an issue with that, but let's get a sign off from @skellock - as he's a bit more fastidious with `.babelrc` files than I.  Also, I just tried copying the file, to verify that is the issue, and it didn't work.  So now I'm not so sure.

**Comment by ryanlntn** [comment]
@GantMan Yeah it's kind of weird because I did a test run locally and was able to commit the change without that error showing up.

**Comment by skellock** [comment]
I wouldn't place the .babelrc in the root if it were me.  The root isn't a node project.

It certainly looks like it cannot read the package.json file.

I'm wondering if it has to do with piping.  Perhaps when we pipe into xargs we lose the current working directory somehow?  

Maybe we have to pass args `--plugin babel-eslint` to that command?  (seems hax)

Are we able to pipe into `npm run lint`?  (seems lame)

**Comment by ryanlntn** [comment]
Per ghooks README:

> Note: The hooks' working directory is relative to the git root (where you have your .git directory). This means that if your package.json is in a subdirectory of your git repo, you'll need to cd into the directory before running any npm scripts. E.g.:

I've updated to the following: `"pre-commit": "if [ -d 'ignite-base' ]; then cd ignite-base; fi; npm run lint"`

**Comment by GantMan** [comment]
where's this PR at?  @skellock  have you verified the full process?   Should I?

**Comment by skellock** [comment]
I was unable to get it to run.  But @ryanlntn since checked in some commits.  I'll have another look on Monday if it's still open.

**Comment by skellock** [comment]
Ok.  Just tested.  Works great.  Green light from me.

**Comment by GantMan** [comment]
Good for you, good for me.

**Pull request merged**